src/eolab/rastertools/hillshade.py

```diff
@@ -105,26 +105,29 @@ def process_file(self, inputfile: str) -> List[str]:
         outdir = Path(self.outputdir)
         output_image = outdir.joinpath(f"{utils.get_basename(inputfile)}-hillshade.tif")
 
-        if self.radius is None:
-            # compute the radius from data range
-            # radius represents the max distance of buildings that can create a hillshade
-            # considering the sun elevation.
-            wmax = None
-            wmin = None
-            with rasterio.open(inputfile) as src:
-                if src.count != 1:
-                    raise ValueError("Invalid input file, it must contain a single band.")
-                for ji, window in src.block_windows(1):
-                    data = src.read(1, masked=True, window=window)
-                    wmax = max(wmax, float(data.max())) if wmax is not None else float(data.max())
-                    wmin = min(wmin, float(data.min())) if wmin is not None else float(data.min())
-
-            delta = int((wmax - wmin) / self.resolution)
-            radius = int(delta / np.tan(np.radians(self.elevation)))
-        else:
-            radius = self.radius
-
-        if radius >= min(self.window_size) / 2:
+        # compute the radius from data range
+        # radius represents the max distance of buildings that can create a hillshade
+        # considering the sun elevation.
+        wmax = None
+        wmin = None
+        with rasterio.open(inputfile) as src:
+            if src.count != 1:
+                raise ValueError("Invalid input file, it must contain a single band.")
+            for ji, window in src.block_windows(1):
+                data = src.read(1, masked=True, window=window)
+                wmax = max(wmax, float(data.max())) if wmax is not None else float(data.max())
+                wmin = min(wmin, float(data.min())) if wmin is not None else float(data.min())
+
+        delta = int((wmax - wmin) / self.resolution)
+        optimal_radius = int(delta / np.tan(np.radians(self.elevation)))
+
+        if self.radius is not None and optimal_radius > self.radius:
+            _logger.warning(f"The radius value is {optimal_radius} exceeding {self.radius} threshold. "
+                            f"Oversized radius affects computation time and so radius is set to {self.radius}. "
+                            "Result may miss some shadow pixels.")
+            optimal_radius = self.radius
```

Comment:
Actually, for code understanding, I will do it the other way around: set `radius` to `optimal_radius` if it is lower.